### PR TITLE
Accurate Character Setup Preview

### PR DESCRIPTION
**Pull request by YotaXP**
Rewrote the silly code that generates the preview icons on the Character Setup screen.  It now builds a dummy mob with the same code that sets up the character when they join, instead of attempting to draw out the character with completely different and unmaintained code.  Also added a 'back' view, and made the icons crisp instead of eye-meltingly blurry.

There is surely a little more overhead, but the results should now be much more accurate, require FAR less maintenance, and should support all existing and future species.  Should be easily portable to other codebases as well, as long as they have the latest `getFlatIcon()` code.

Before and after shot

I didn't bother doing anything special for the silicons since their appearance is determined mid-round instead of before, but I will if there's enough of a demand for it.

:cl:
tweak: Spruced up the preview icons on the Character Setup dialog.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
Holy shit

**Comment by tkdrg** [comment]
YES
I LOVE YOU YOTA

**Comment by AnturK** [comment]
+:100:  Love

**Comment by drynwyn** [comment]
well done

**Comment by as334** [comment]
You are literally my favourite person ever.

**Comment by GunHog** [comment]
It is like an upgrade to HD graphics!

**Comment by Iamgoofball** [comment]
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: 
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100:

**Comment by AnturK** [comment]
Check if it triggers "Captain joined" though. I'm pretty sure i didn't add any checks for that in outfit code.

File: code/modules/mob/new_player/preferences_setup.dm [hunk 1]
```diff
@@ -20,407 +20,62 @@
...
-	del(clothes_s)
\ No newline at end of file
+				preview_icon = icon('icons/mob/robots.dmi', "robot", SOUTH)
+				preview_icon.Scale(64, 64)
+				return
+
+	// Set up the dummy for its photoshoot
+	var/mob/living/carbon/human/dummy/mannequin = new()
+	mannequin.status_flags |= GODMODE // Why not?
+	copy_to(mannequin)
```
Comment (MrPerson) [inline code/modules/mob/new_player/preferences_setup.dm]: I'm pretty sure dummies already have GODMODE set by default but ok

Comment (RemieRichards) [inline code/modules/mob/new_player/preferences_setup.dm]: Yeah, they do.

**Comment by Anonus** [comment]
:+1: holy shit it's amazing I love it sign me the fuck up

**Comment by PKPenguin321** [comment]
if you set them to plasmamen does the dummy burst into flames on the setup menu because that would be funny

**Comment by RemieRichards** [comment]
@PKPenguin321 they exist in nullspace (essentially outside of the gameworld) and there's no oxygen there, so no :P

**Comment by WJohn** [comment]
IS THIS WHAT PUTTING ON GLASSES IS LIKE?

EVERYTHING IS HD NOW.

**Comment by Incoming5643** [comment]
Never even noticed female clothing icon generation didn't work on previews cause it was so damn blurry.

:+1:

File: code/modules/mob/new_player/preferences_setup.dm (hunk 1 above)
Comment (YotaXP) [inline code/modules/mob/new_player/preferences_setup.dm]: Good to know!  I'll remove that line if I push any more commits.

**Comment by YotaXP** [comment]
The mob is poolable if we override its `Life()` to do nothing, but the mob shows up in verb lists if we don't delete it quickly. (Such as a ghosts's "Jump to Mob") So unless someone knows of a trick to hide them, we'll have to just create and delete them over and over. (I tried making it invisble and clearing its name.)  Dunno how much it matters anyway, though, since none of the equipment it puts on is pooled yet anyway.

**Comment by duncathan** [comment]
Lol the old plasmamen icons

**Comment by duncathan** [comment]
Wrong button

**Comment by daman997** [comment]
ye :+1:

**Comment by bear1ake** [comment]
COOL :+1: :x: :100:

**Comment by Cheridan** [comment]
a hero and a true american

**Comment by YotaXP** [comment]
Just gonna leave a note here in case any other codebases try to use this.  You'll need the fix provided by #13224 as well.

**Pull request merged**